app/javascript/src/NewService/components/FormTemplate.jsx

```diff
@@ -0,0 +1,98 @@
+  data-selected-value={namespace.id === 'service_namespace' ? props.selectedNamespace : ''}>
+  {props.namespaces && renderSelectOptions(props.namespaces)}
+</select>
+
+const renderServiceDiscoveriItems = (props, serviceDiscoveryProps) => {
+  return (
+    <React.Fragment>
+      <li id="service_name_input" className="string required">
+        { renderLabel(serviceDiscoveryProps.serviceDiscoverySelect.namespace, false) }
+        { renderSelect(serviceDiscoveryProps.serviceDiscoverySelect.namespace, props) }
+      </li>
+      <li>
+        { renderLabel(serviceDiscoveryProps.serviceDiscoverySelect.projects, false) }
+        { renderSelect(serviceDiscoveryProps.serviceDiscoverySelect.projects, props) }
+      </li>
+    </React.Fragment>
+  )
+}
+
+const renderNewServiceItems = () => {
+  return (
+    <React.Fragment>
+      <li id="service_name_input" className="string required">
+        {renderLabel({htmlFor: 'service_name', label: 'Name'}, true)}
+        <input maxLength="255" id="service_name" type="text" name="service[name]" autoFocus />
+      </li>
+      <li id="service_system_name_input" className="string required">
+        {renderLabel({htmlFor: 'service_system_name', label: 'System name'}, true)}
+        <input maxLength="255" id="service_system_name" type="text" name="service[system_name]" />
+        <p className="inline-hints">Only ASCII letters, numbers, dashes and underscores are allowed.</p>{/* TODO: Get this from rails */}
+      </li>
+      <li id="service_description_input" className="text optional">
+        {renderLabel({htmlFor: 'service_description', label: 'Description'}, false)}
+        <textarea rows="3" id="service_description" name="service[description]">
+        </textarea>
+      </li>
+    </React.Fragment>
+  )
+}
+
+const FormTemplate = (props: FormTemplateProps) => {
+  const {isFormDiscovery, isVisible, formActionPath} = props
+  const serviceDiscoveryProps = getServiceDiscoveryProps()
+
+  return (
+    <form className={`formtastic service ${isVisible ? '' : 'is-hidden'}`} id={isFormDiscovery ? 'service_discovery' : 'new_service'} action={formActionPath} acceptCharset="UTF-8" method="post">
+      <input name="utf8" type="hidden" value="✓"/>
+      <CSRFToken />
+      { isFormDiscovery && <input value={serviceDiscoveryProps.hiddenServiceDiscoveryInput.value} type="hidden" name={serviceDiscoveryProps.hiddenServiceDiscoveryInput.name} id={serviceDiscoveryProps.hiddenServiceDiscoveryInput.id} /> }
+      <fieldset className="inputs" name="Service">
+        <legend><span>Service</span></legend>
+        <ol>
+          { isFormDiscovery ? renderServiceDiscoveriItems(props, serviceDiscoveryProps) : renderNewServiceItems() }
+        </ol>
+      </fieldset>
+      <input type="submit" name="commit" value={isFormDiscovery ? 'Create Service' : 'Add API'} className="important-button create"/>
```

Comment:
Maybe we could use "Add API" for both.

Reply:
"Create Service" can be confusing for service discovery, after all the (OpenShift) Service already exists. We are actually adding the API for management with 3scale.

Reply:
I suppose we shouldn't change this text for the same reason explained [here](https://github.com/3scale/porta/pull/763/files?file-filters%5B%5D=.feature&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.jsx&file-filters%5B%5D=.slim#r282824691): It'd need docs update.